### PR TITLE
Add tests from Diffblue Deeptest

### PR DESCRIPTION
**Pull request by paulkennethkent**
Existing Coverage:  68.51% 
Coverage Increase:  6.26% 
New total coverage: 74.78% 
Total tests: 148
Validated Tests: 31

### CODE REVIEW

File: src/test/java/spark/utils/StringUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 397 branch to line 397
+   *  - conditional line 397 branch to line 398
+   *  - conditional line 398 branch to line 398
+   *  - conditional line 401 branch to line 402
+   *  - conditional line 402 branch to line 402
+   */
+
```
Comment (paulkennethkent) [inline src/test/java/spark/utils/StringUtilsTest.java:288]: This is one of my favourite tests

Comment (sudo-jaa) [reply src/test/java/spark/utils/StringUtilsTest.java:288]: This test also fills me with glee.

Comment (peterschrammel) [reply src/test/java/spark/utils/StringUtilsTest.java:288]: Is this a bug in the method tested?

File: src/test/java/spark/utils/urldecoding/TypeUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+   *     - conditional line 187 branch to line 190
+   *     - conditional line 192 branch to line 195
+   */
+
+  @Test
+  public void toStringInput1NegativeOutputNotNull() {
+
+    // Arrange
+    final byte[] bytes = {(byte)0};
+    final int base = -2_147_483_648;
```
Comment (peterschrammel) [inline src/test/java/spark/utils/urldecoding/TypeUtilTest.java:213]: A negative base? Interesting...

File: src/test/java/spark/utils/ClassUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  /* testedClasses: ClassUtils */
+
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 254 branch to line 255
+   */
+
```
Comment (paulkennethkent) [inline src/test/java/spark/utils/ClassUtilsTest.java:35]: Create a jira ticket about what isn't covered in this method

File: src/test/java/spark/utils/ClassUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 231 branch to line 231
+   *  - conditional line 231 branch to line 235
+   */
+
+  @Test
+  public void resolvePrimitiveClassNameInputNotNullOutputNull() {
```
Comment (paulkennethkent) [inline src/test/java/spark/utils/ClassUtilsTest.java:57]: Check what is not covered

File: src/test/java/spark/utils/urldecoding/TypeUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 165 branch to line 166
+   *  - conditional line 169 branch to line
+   */
+
+  @Test
+  public void parseIntInputNotNullPositiveNegativeZeroOutputZero() {
```
Comment (paulkennethkent) [inline src/test/java/spark/utils/urldecoding/TypeUtilTest.java:89]: Bug

File: src/test/java/spark/utils/MimeParseTest.java [hunk 6]
```diff
@@ -4,11 +4,200 @@
...
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+
 
 public class MimeParseTest {
 
+  @Rule public ExpectedException thrown = ExpectedException.none();
```
Comment (sudo-jaa) [inline src/test/java/spark/utils/MimeParseTest.java]: Are we happy keeping this level of reflection for tests that access private methods?

**Review by sudo-jaa — APPROVED** [review]

File: src/test/java/spark/utils/MimeParseTest.java [hunk 7]
```diff
@@ -4,11 +4,200 @@
...
+   * This test covers `String toString' block 16 (line 34)
+   * This test covers `String toString' block 17 (line 34)
+   * This test covers `String toString' block 18 (line 37)
+   * This test covers `String toString' block 34 (line 37)
+   * This test covers `String toString' block 35 (line 37)
+   * This test covers `String toString' block 36 (line 37)
+   *
+   */
+
+  @Test
```
Comment (petersear) [inline src/test/java/spark/utils/MimeParseTest.java]: Is there another test that goes through the "for" loop?

Comment (petersear) [reply src/test/java/spark/utils/MimeParseTest.java]: Oh yes, the next test does ;-)

**Review by petersear — APPROVED** [review]

File: src/test/java/spark/utils/MimeParseTest.java (hunk 6 above)
Comment (paulkennethkent) [reply src/test/java/spark/utils/MimeParseTest.java]: I will remove this test class just to be super clean

File: src/test/java/spark/utils/ClassUtilsTest.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 254 branch to line 255
+   */
+
+  @Test
+  public void classPackageAsResourcePathInputNullOutputNotNull() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/ClassUtilsTest.java:38]: :+1:

File: src/test/java/spark/utils/ClassUtilsTest.java [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 231 branch to line 231
+   *  - conditional line 231 branch to line 235
+   */
+
+  @Test
+  public void resolvePrimitiveClassNameInputNotNullOutputNull() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/ClassUtilsTest.java:58]: :+1:

File: src/test/java/spark/utils/ClassUtilsTest.java [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 231 branch to line 235
+   */
+
+  @Test
+  public void resolvePrimitiveClassNameInputNullOutputNull() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/ClassUtilsTest.java:77]: :+1:

File: src/test/java/spark/utils/GzipUtilsTest.java [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 91 branch to line 92
+   */
+
+  @Test
+  public void testInputNullOutputFalse()
+      throws NoSuchMethodException, IllegalAccessException, InvocationTargetException {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/GzipUtilsTest.java:43]: :-1: Too much reflection (unless reflection has been created correctly - i.e. no public constructor, in which case: :+1:

File: src/test/java/spark/utils/StringUtilsTest.java [hunk 12]
```diff
@@ -0,0 +1,309 @@
...
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 198 branch to line 199
+   *  - conditional line 199 branch to line 199
+   *  - conditional line 200 branch to line 203
+   */
+
+  @Test
+  public void applyRelativePathInputNotNullNotNullOutputNotNull() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/StringUtilsTest.java:43]: :+1:

File: src/test/java/spark/utils/StringUtilsTest.java [hunk 13]
```diff
@@ -0,0 +1,309 @@
...
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 198 branch to line 205
+   */
+
+  @Test
+  public void applyRelativePathInputNotNullNotNullOutputNotNull2() {
+
+    // Arrange
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/StringUtilsTest.java:64]: :+1:

File: src/test/java/spark/utils/StringUtilsTest.java [hunk 14]
```diff
@@ -0,0 +1,309 @@
...
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 198 branch to line 199
+   *  - conditional line 199 branch to line 199
+   *  - conditional line 200 branch to line 201
+   */
+
+  @Test
+  public void applyRelativePathInputNotNullNotNullOutputNotNull3() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/StringUtilsTest.java:85]: :+1:

File: src/test/java/spark/utils/StringUtilsTest.java [hunk 15]
```diff
@@ -0,0 +1,309 @@
...
+   *  - conditional line 160 branch to line 163
+   *  - spark/utils/StringUtils.java:164: loop: 1 iterations
+   *  - iteration 1
+   *     - conditional line 165 branch to line 165
+   *     - conditional line 166 branch to line 167
+   */
+
+  @Test
+  public void deleteAnyInputNotNullNotNullOutputNotNull() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/StringUtilsTest.java:110]: :+1:

File: src/test/java/spark/utils/StringUtilsTest.java [hunk 16]
```diff
@@ -0,0 +1,309 @@
...
+   *  - conditional line 160 branch to line 163
+   *  - spark/utils/StringUtils.java:164: loop: 1 iterations
+   *  - iteration 1
+   *     - conditional line 165 branch to line 165
+   *     - conditional line 166 branch to line 164
+   */
+
+  @Test
+  public void deleteAnyInputNotNullNotNullOutputNotNull2() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/StringUtilsTest.java:135]: :+1:

File: src/test/java/spark/utils/StringUtilsTest.java [hunk 17]
```diff
@@ -0,0 +1,309 @@
...
+   * This test case covers:
+   *  - conditional line 313 branch to line 316
+   *  - conditional line 316 branch to line 319
+   *  - conditional line 320 branch to line 321
+   *  - conditional line 321 branch to line
+   */
+
+  @Test
+  public void delimitedListToStringArrayInputNotNullNotNullNotNullOutput0() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/StringUtilsTest.java:158]: :+1:

File: src/test/java/spark/utils/StringUtilsTest.java [hunk 18]
```diff
@@ -0,0 +1,309 @@
...
+   *  - conditional line 320 branch to line 321
+   *  - spark/utils/StringUtils.java:321: loop: 1 iterations
+   *  - iteration 1
+   *     - conditional line 322 branch to line 322
+   */
+
+  @Test
+  public void delimitedListToStringArrayInputNotNullNotNullNotNullOutput1() {
+
+    // Arrange
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/StringUtilsTest.java:185]: :+1:

File: src/test/java/spark/utils/StringUtilsTest.java [hunk 19]
```diff
@@ -0,0 +1,309 @@
...
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 313 branch to line 316
+   *  - conditional line 316 branch to line 317
+   */
+
+  @Test
+  public void delimitedListToStringArrayInputNotNullNullNotNullOutput1() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/StringUtilsTest.java:206]: :+1:

File: src/test/java/spark/utils/StringUtilsTest.java [hunk 20]
```diff
@@ -0,0 +1,309 @@
...
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 313 branch to line 314
+   */
+
+  @Test
+  public void delimitedListToStringArrayInputNullNullNotNullOutput0() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/StringUtilsTest.java:227]: :+1:

File: src/test/java/spark/utils/StringUtilsTest.java [hunk 21]
```diff
@@ -0,0 +1,309 @@
...
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 97 branch to line 97
+   */
+
+  @Test
+  public void isNotEmptyInputNotNullOutputFalse() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/StringUtilsTest.java:249]: :+1:

File: src/test/java/spark/utils/StringUtilsTest.java [hunk 22]
```diff
@@ -0,0 +1,309 @@
...
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 97 branch to line 97
+   */
+
+  @Test
+  public void isNotEmptyInputNotNullOutputTrue() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/StringUtilsTest.java:268]: :+1:

File: src/test/java/spark/utils/StringUtilsTest.java [hunk 23]
```diff
@@ -0,0 +1,309 @@
...
+   *  - conditional line 397 branch to line 397
+   *  - conditional line 397 branch to line 398
+   *  - conditional line 398 branch to line 398
+   *  - conditional line 401 branch to line 402
+   *  - conditional line 402 branch to line 402
+   */
+
+  @Test
+  public void removeLeadingAndTrailingSlashesFromInputNotNullOutputNotNull() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/StringUtilsTest.java:291]: :+1:

File: src/test/java/spark/utils/StringUtilsTest.java [hunk 24]
```diff
@@ -0,0 +1,309 @@
...
+   * This test case covers:
+   *  - conditional line 397 branch to line 398
+   *  - conditional line 398 branch to line 398
+   *  - conditional line 401 branch to line 402
+   *  - conditional line 402 branch to line 402
+   */
+
+  @Test
+  public void removeLeadingAndTrailingSlashesFromInputNotNullOutputNotNull2() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/StringUtilsTest.java:313]: :+1:

File: src/test/java/spark/utils/urldecoding/TypeUtilTest.java [hunk 25]
```diff
@@ -0,0 +1,253 @@
...
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 222 branch to line 222
+   *  - conditional line 222 branch to line 225
+   */
+
+  @Test
+  public void convertHexDigitInputPositiveOutputPositive() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/urldecoding/TypeUtilTest.java:42]: :+1:

File: src/test/java/spark/utils/urldecoding/TypeUtilTest.java [hunk 26]
```diff
@@ -0,0 +1,253 @@
...
+   * This test covers `int parseInt(String, int, int)' block 5 (line 166)
+   * This test covers `int parseInt(String, int, int)' block 6 (line 169)
+   * This test covers `int parseInt(String, int, int)' block 7 (line 169)
+   * This test covers `int parseInt(String, int, int)' block 9 (line 170)
+   *
+   */
+
+  @Test
+  public void parseIntInputNotNullNegativeNegativeZeroOutputStringIndexOutOfBoundsException() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/urldecoding/TypeUtilTest.java:67]: :+1:

File: src/test/java/spark/utils/urldecoding/TypeUtilTest.java [hunk 27]
```diff
@@ -0,0 +1,253 @@
...
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 165 branch to line 166
+   *  - conditional line 169 branch to line
+   */
+
+  @Test
+  public void parseIntInputNotNullPositiveNegativeZeroOutputZero() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/urldecoding/TypeUtilTest.java:90]: :+1:

File: src/test/java/spark/utils/urldecoding/TypeUtilTest.java [hunk 28]
```diff
@@ -0,0 +1,253 @@
...
+   * This test covers `int parseInt(String, int, int)' block 3 (line 165)
+   * This test covers `int parseInt(String, int, int)' block 6 (line 169)
+   * This test covers `int parseInt(String, int, int)' block 7 (line 169)
+   * This test covers `int parseInt(String, int, int)' block 9 (line 170)
+   *
+   */
+
+  @Test
+  public void parseIntInputNotNullPositivePositiveZeroOutputStringIndexOutOfBoundsException() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/urldecoding/TypeUtilTest.java:116]: :+1:

File: src/test/java/spark/utils/urldecoding/TypeUtilTest.java [hunk 29]
```diff
@@ -0,0 +1,253 @@
...
+   * This test case covers:
+   *  - spark/utils/urldecoding/TypeUtil.java:236: loop: 1 iterations
+   *  - iteration 1
+   *     - conditional line 239 branch to line 240
+   *     - conditional line 244 branch to line 247
+   */
+
+  @Test
+  public void toHexStringInput1ZeroPositiveOutputNotNull() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/urldecoding/TypeUtilTest.java:141]: :+1:

File: src/test/java/spark/utils/urldecoding/TypeUtilTest.java [hunk 30]
```diff
@@ -0,0 +1,253 @@
...
+   * This test case covers:
+   *  - spark/utils/urldecoding/TypeUtil.java:236: loop: 1 iterations
+   *  - iteration 1
+   *     - conditional line 239 branch to line 240
+   *     - conditional line 244 branch to line 245
+   */
+
+  @Test
+  public void toHexStringInput1ZeroPositiveOutputNotNull2() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/urldecoding/TypeUtilTest.java:165]: :+1:

File: src/test/java/spark/utils/urldecoding/TypeUtilTest.java [hunk 31]
```diff
@@ -0,0 +1,253 @@
...
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *
+   */
+
+  @Test
+  public void toHexStringInputNullZeroZeroOutputNotNull() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/urldecoding/TypeUtilTest.java:186]: :+1:

File: src/test/java/spark/utils/urldecoding/TypeUtilTest.java [hunk 32]
```diff
@@ -0,0 +1,253 @@
...
+   * This test case covers:
+   *  - spark/utils/urldecoding/TypeUtil.java:184: loop: 1 iterations
+   *  - iteration 1
+   *     - conditional line 187 branch to line 190
+   *     - conditional line 192 branch to line 195
+   */
+
+  @Test
+  public void toStringInput1NegativeOutputNotNull() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/urldecoding/TypeUtilTest.java:210]: :+1:

File: src/test/java/spark/utils/urldecoding/TypeUtilTest.java [hunk 33]
```diff
@@ -0,0 +1,253 @@
...
+   * This test case covers:
+   *  - spark/utils/urldecoding/TypeUtil.java:184: loop: 1 iterations
+   *  - iteration 1
+   *     - conditional line 187 branch to line 190
+   *     - conditional line 192 branch to line 193
+   */
+
+  @Test
+  public void toStringInput1NegativeOutputNotNull2() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/urldecoding/TypeUtilTest.java:233]: :+1:

File: src/test/java/spark/utils/urldecoding/TypeUtilTest.java [hunk 34]
```diff
@@ -0,0 +1,253 @@
...
+   * This test case covers:
+   *  - spark/utils/urldecoding/TypeUtil.java:184: loop: 1 iterations
+   *  - iteration 1
+   *     - conditional line 187 branch to line 188
+   *     - conditional line 192 branch to line 195
+   */
+
+  @Test
+  public void toStringInput1PositiveOutputNotNull() {
+
```
Comment (JohnLBergqvist) [inline src/test/java/spark/utils/urldecoding/TypeUtilTest.java:256]: :+1:

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]
Possibly one of the tests can be discarded, if it's using reflection unnecessarily

**Pull request closed without merging**